src/lazy/expanded/mod.rs

```diff
@@ -289,6 +289,44 @@ impl<'top> EncodingContextRef<'top> {
             .clone_macro_with_name("values")
             .expect("`values` macro in system macro table")
     }
+
+    #[cfg(feature = "lazy-source-location")]
+    pub fn location(&self, value_start: usize) -> Option<(usize, usize)> {
+        match unsafe { &*self.io_buffer_source.get() } {
+            IoBufferSource::IoBuffer(ref buffer) => Some((buffer.row(), buffer.column())),
+            IoBufferSource::Reader(handle) => {
+                let prev_newline = handle.prev_newline_offset();
+                let current_row = handle.row();
+
+                if prev_newline == 0 {
+                    // If no newlines have been encountered yet, we're on the first row
+                    // The column is the start position of the value
+                    Some((1, value_start))
+                } else {
+                    // If newlines have been encountered
+                    if value_start > prev_newline {
+                        // If the value starts after the last newline,
+                        // we're on the current row and can calculate the column
+                        Some((current_row, value_start - prev_newline))
+                    } else {
+                        // If the value starts before or at the last newline,
+                        // we need to find the correct row and column
+                        if let Some(last_newline_pos) = handle.newlines().iter().enumerate()
```

Comment:
This branch is not used for `IonSlice`, right? That suggests to me that things aren't encapsulated the right way. It doesn't necessarily need to be addressed now, but this could easily become a friction point if we're trying to refactor things at some point in the future (either related to source location or related to `IoBufferHandle`, etc).

Reply:
https://github.com/amazon-ion/ion-rust/issues/944